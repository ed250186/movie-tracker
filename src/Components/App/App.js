import React, { Component } from "react";
import "./App.css";
import { nowPlaying } from "../../apiCalls/apiCalls";

class App extends Component {
  constructor() {
    super();
    this.state = {
      movies: []
    };
  }
  componentDidMount() {
    nowPlaying()
      .then(movies => this.setState({ movies: movies}))
      .catch(this.setState({ error: 'Error fetching data' }));
    console.log('did this work')
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <p>
            Edit <code>src/App.js</code> and save to reload.
          </p>
          <a
            className="App-link"
            href="https://reactjs.org"
            target="_blank"
            rel="noopener noreferrer"
          >
            Learn React
          </a>
        </header>
      </div>
    );
  }
}

export default App;
