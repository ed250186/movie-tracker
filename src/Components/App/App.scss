* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

html,
body,
#root {
  height: 100vh;
  width: 100vw;
}

.App {
  display: flex;
  min-height: 100%;
  width: 100%;
  main {
    height: 100%;
    width: 90%
  }
}

button,
.button {
  background: #485981;
  border: none;
  border-radius: 50px;
  color: #DCE3EF;
}
button:hover,
.button:hover {
  background: #FF1070;
}