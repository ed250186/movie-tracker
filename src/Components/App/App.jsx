import React, { Component } from "react";
import "./App.scss";
import Aside from '../Aside/Aside.jsx'
import Header from '../Header/Header.jsx'
import MovieContainer from '../../Containers/MovieContainer/MovieContainer.jsx'
// import { connect }  from 'react-redux';
import SignIn from '../SignIn/SignIn';
import SignUp from '../SignUp/SignUp'


class App extends Component {

  render() {
    return (
      <div className="App">
        <Aside />
        <main>
          <Header />
          <MovieContainer />
          <SignIn />
        </main>
        {/* <SignUp/> */}
      </div>
    );
  }
}


export default App;
