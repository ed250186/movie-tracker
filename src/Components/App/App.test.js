import React from 'react';
import {shallow} from 'enzyme';
import App from './App';

describe('App', () => {
  let wrapper, instance;

  beforeEach(() => {
    wrapper = shallow(<App/>);
    instance = wrapper.instance()
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render people home page', () => {
    global.window = Object.create(window);
    let url = 'http://localhost:3000/';
    Object.defineProperty(window, 'location', {
      value: {
       href: url
      }
    });
    expect(instance.main()).toMatchSnapshot()
  })

})