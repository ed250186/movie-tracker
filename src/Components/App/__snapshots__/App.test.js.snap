// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<div
  className="App"
>
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(SignIn)",
        "type": [Function],
      }
    }
    exact={true}
    path="/signin"
  />
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(SignUp)",
        "type": [Function],
      }
    }
    exact={true}
    path="/signup"
  />
</div>
`;

exports[`App should render people home page 1`] = `
<React.Fragment>
  <Aside />
  <main>
    <Memo(ConnectFunction) />
    <Memo(ConnectFunction) />
  </main>
</React.Fragment>
`;
